app/controllers/cookbooks_controller.rb

```diff
@@ -165,19 +165,16 @@ def unfollow
   def deprecate
     authorize! @cookbook
 
-    replacement_cookbook = Cookbook.with_name(
-      cookbook_deprecation_params[:replacement]
-    ).first!
+    replacement_cookbook_name = cookbook_deprecation_params[:cookbook][:replacement]
```

Comment:
Moved the replacement cookbook query into `Cookbook.deprecate` so we only need the name string here in the controller.